src/niri.rs

```diff
@@ -2077,6 +2106,50 @@ impl Niri {
             .cloned()
     }
 
+    pub fn output_by_name(&self, name: Option<&str>) -> Option<Output> {
+        let Some(name) = name else {
+            return None;
+        };
+        self.global_space
+            .outputs()
+            .find(|output| &output.name() == name)
+            .cloned()
+    }
+
+    pub fn find_output_and_workspace_index(
+        &self,
+        workspace_reference: WorkspaceReference,
+    ) -> Option<(Option<Output>, usize)> {
+        if let WorkspaceReference::Index(index) = workspace_reference {
+            return Some((None, index.saturating_sub(1) as usize));
+        };
+
+        let WorkspaceReference::Name(workspace_name) = workspace_reference else {
+            return None;
+        };
```

Comment:
You can do this with a match:
```suggestion
        let workspace_name = match workspace_reference {
            WorkspaceReference::Index(index) => {
                return Some((None, index.saturating_sub(1) as usize));
            }
            WorkspaceReference::Name(name) => name,
        };
```